pkg/controllers/csi/driver/server.go

```diff
@@ -62,39 +61,36 @@ type Server struct {
 var _ csi.IdentityServer = &Server{}
 var _ csi.NodeServer = &Server{}
 
-func NewServer(opts dtcsi.CSIOptions, db metadata.Access) *Server {
+func NewServer(opts dtcsi.CSIOptions) *Server {
 	return &Server{
 		opts:    opts,
 		fs:      afero.Afero{Fs: afero.NewOsFs()},
 		mounter: mount.New(""),
-		db:      db,
 		path:    metadata.PathResolver{RootDir: opts.RootDir},
 	}
 }
 
-func (svr *Server) SetupWithManager(mgr ctrl.Manager) error {
-	return mgr.Add(svr)
+func (srv *Server) SetupWithManager(mgr ctrl.Manager) error {
+	return mgr.Add(srv)
 }
 
-func (svr *Server) Start(ctx context.Context) error {
-	defer metadata.LogAccessOverview(svr.db)
-
-	endpoint, err := url.Parse(svr.opts.Endpoint)
+func (srv *Server) Start(ctx context.Context) error {
+	endpoint, err := url.Parse(srv.opts.Endpoint)
 	addr := endpoint.Host + endpoint.Path
```

Comment:
I think it was before like this, but code will panic if Parse return an err, 
```
invalid operation: operator + not defined on nil

Go build failed.
```

we should move check error before we do:
`addr := endpoint.Host + endpoint.Path` somethig like it
```
	endpoint, err := url.Parse(srv.opts.Endpoint)
	
	if err != nil {
		return fmt.Errorf("failed to parse endpoint '%s': %w", srv.opts.Endpoint, err)
	}
	
	addr := endpoint.Host + endpoint.Path

```

Reply:
done